jupyter_collaboration/app.py

```diff
@@ -59,5 +64,7 @@ def initialize_handlers(self):
             [
                 (r"/api/yjs/roomid/(.*)", YDocRoomIdHandler),
                 (r"/api/yjs/(.*)", YDocWebSocketHandler),
+                (r"/api/collaboration/room/(.*)", YDocWebSocketHandler),
+                (r"/api/collaboration/session/(.*)", DocSessionHandler),
```

Comment:
I don't think we should grow the API surface.
I think we should use the already existing `PUT /api/yjs/roomid/{path}` endpoint. It could return a value that is a combination of the file ID, the document type and format, and the session ID. That way, the frontend doesn't have to ask for a session ID, all it asks for is a room ID. When it tries to connect to the WebSocket with a room ID that doesn't exit (because the server session has changed), the room cannot be found and the frontend asks for the room ID again, discarding the shared document that it may have.

Reply:
By combination, I don't mean necessarily concatenating multiple attributes, it can be a UUID and the server keeps a mapping of this UUID to the attributes.

Reply:
It is easy for the frontend to know that the session has changed: it fails to connect with its room ID. It can now ask for the room ID again, if that fails it's because the document doesn't exist, otherwise it's because the server has restarted, and in that case it can tell the user to not use the tab.

Reply:
Implicit deduction is never a good path for code evolution, maintenance and understanding.